content/post/ml_talks_moscow.md

```diff
@@ -0,0 +1,163 @@
+}
+</style>
+
+Then, after everybody woke up slowly, it was time for our CEO, [Eiso Kant](https://twitter.com/eisokant), to launch the main talks series. These lasted 45min each with time for Q&A. Furthermore, 2 lightning talks of 15min occurred between the main ones to address more specific topics.
+
+# Main talks
+
+### Statistical Analysis of Computer Program Text, _Charles Sutton_
+
+**_"Source code is a means of human communication"_**, with this first formula, the professor at University of Edinburgh, Charles Sutton, couldn't better start the day. He next laid out his statistical approach to analyze source code texts. In order to extract from scripts what he called, _implicit knowledge_, he introduced three innovative software engineering tools inspired from machine learning and natural language processing (NLP) techniques :
+
+* **Naturalize**, a probabilistic language model for source code which learns local coding conventions. It suggests renaming or reformatting changes so that your code would become more consistent.
+* **HAGGIS, Mining idioms for code** a system that learns local recurring syntactic patterns, which we call idioms, using a nonparametric Bayesian tree substitution grammar (TSG).
+* **Probabilistic API Miner (PAM)**, a near parameter-free probabilistic algorithm for mining the most interesting API patterns from a list of API call sequences. It resolves fundamental statistical pathologies like the formation of redundant or spurious sequences.
+</br>
+</br>
+
+<img src="/post/ml_talks_moscow/charles.jpeg" id="cs" alt="Photo of talking Charles Sutton">
+<style>
+#cs {
+  width: 600px;
+  height: 300px;
+  object-fit: contain;
+}
+</style>
+
+
+### Similarity of GitHub repositories by source code identifiers, _Vadim Markovstev_
+
+<img src="/post/ml_talks_moscow/vadim.jpg" id="vm" alt="Photo of talking Vadim Markovstev">
+<style>
+#vm {
+  width: 500px;
+  height: 250px;
+  object-fit: contain;
+}
+</style>
+
+### Probabilistic Programming for Mere Mortals, _Vitaly Khudobakhshov_
+
+In his talk, Vitaly presented an review of an emerging topic which is probabilistic programming. He described programming languages problems where this field of research could offer some solutions.
+
+Then, he started a discussion about how we could implement probabilistic programming language and how we could embed probabilistic programming capabilities into a general purpose programming language.
+
+<img src="/post/ml_talks_moscow/vitaly.jpg" id="vk" alt="Photo of talking Vitaly Khudobakhshov">
+<style>
+#vk {
+  width: 500px;
+  height: 250px;
+  object-fit: contain;
+}
+</style>
+
+### Sequence Learning and modern RNNs, _Grigory Sapunov_
+
+Grigory started his talk with a tiny intro into RNN, LSTM and GRU
+
+<img src="/post/ml_talks_moscow/grigory.jpg" id="gs" alt="Photo of talking Grigory Sapunov">
+<style>
+#gs {
+  width: 500px;
+  height: 250px;
+  object-fit: contain;
+}
+</style>
+
+### Neural Complete project, _Pascal Van Kooten
```

Comment:
Indeed,
Furthermore, I wrote a choose a complete random convention to write those subtitles